### ISSUE TITLE
Requisitos de segurança para uma aplicação

### ISSUE THREAD

**Issue by williamnaltchadjian** [grupozap/guild-paranoias#2]
## O que é a demanda?

Por um questão de segurança e boas praticas precisamos discutir quais são os requisitos de segurança para uma aplicação.

## Quais são os benefícios?

Ter uma politica para definir os requisitos de segurança para uma aplicação.

## O que faremos?

- [ ] Coletar Sugestões da guilda
- [ ] Discutir com a engenharia
- [ ] Criar Documento com definições

**Comment by williamnaltchadjian** [comment]
Segue a minha sugestão:

- Utilizar certificado  tanto para aplicações internas como externas.(Importante utilizar internamente pois precisamos que todo input no servidor seja criptografado).
- Utilizar waf para aplicações públicas.
- Todo sistema interno deve ter autenticação SSO.
- Todo o fluxo de CI deve ter sonarqube para avaliar a qualidade do código.
- Todas as aplicações que rodam no kubernetes e EC2 utilizando a conta do vivareal, devem usar roles.

**Comment by luizcasali** [comment]
acho interessante criar níveis de criticidade e depois classificar cada aplicação em um desses níveis, para não haver uma regra específica pra cada e virar um monstro.
ex: criticidade:
- alta
- média
- baixa

ou criticidade 1, 2, 3 ...

ai definimos cada níveis, depois colocamos a aplicação em um desses níveis.

**Comment by Lucas-L-Alcantara** [comment]
Concordo com o @luizcasali sobre criar níveis de criticidade e pra isso devemos definir alguns crítérios de risco, como: exposição (se privada ou pública), se possui área autenticada (área de cliente ou de administração), se utiliza banco de dados... 

Depois disso, encaixar em cada **"Paranóia Level"** os critérios que estamos listando.

Um tempo atras, numa discussão geral sobre qualidade, inclui alguns itens de segurança, pois como estamos pregando desde o início do novo modelo da área, **não há qualidade sem segurança:**
https://github.com/orgs/grupozap/teams/engineering/discussions/2/comments/10

    Utilizar HTTPS
    Utilizar WAF
    Evitar "Information disclosure"
    Autenticação (Definir Complexidade de senhas + Evitar "Login Information Disclosure")
    Uso seguro de cookies
    Proteção de dados (Criptografia por ex)
    Opções seguras para o Header, como X-frame-Options
    Gerenciamento seguro de sessões
    Minimamente não possuir vulnerabilidades citadas na OWASP TOP 10. Nisso nós de Security podemos sempre apoiar com scans desde o período de testes.

Na mesma discussão eu sugeri a consulta à documentação de melhores práticas da OWASP (https://www.owasp.org/images/b/b3/OWASP_SCP_v1.3_pt-BR.pdf), poi ela tras algumas coisas como, proteção a dados, gerenciamento de memória e outras coisas que podem virar critérios nossos.

Acho importante o que o @williamnaltchadjian disse sobre todos os repos estarem configurados para o Sonar, pois lá já temos algumas coisas de SEC (OWASP top 10 por ex). Além do Sonar, incluir verificação SAST (Static Application Security Testing) como o Bandit e DAST (Dunamic Application Security Testing) com o Arachni,  além de verificação de vulnerabilidade em dependências na esteira de desenvolvimento.

Utilizar subdependências  ao invés de importar a lib inteira no código. Isso reduz a probabilidade de vulnerabilidades, assim com utilizar versão mais recente das linguagens ( vide https://nvd.nist.gov/vuln/detail/CVE-2019-9948).

Algo MUITO IMPORTANTE para sec que já virou OWASP TOP 10: **LOGS**

**Comment by Lucas-L-Alcantara** [comment]
Poderiamos considerar também alguns aspectos lendo o dockerfile, se está sendo utilizada imagem oficial ou não, se está dentro da nossa esteira... (inclusive o @phspagiari estava fazendo um script para identificar isso)

**Comment by rodrigozrusso** [comment]
@Lucas-L-Alcantara https://sysdig.com/blog/20-docker-security-tools/
